src/FffGcodeWriter.cpp

```diff
@@ -851,6 +851,8 @@ LayerPlan& FffGcodeWriter::processLayer(const SliceDataStorage& storage, LayerIn
                 
                 // path optimization is currently broken when using gyroid infill
                 disable_path_optimisation = disable_path_optimisation || mesh.settings.get<EFillMethod>("infill_pattern") == EFillMethod::GYROID;
+                // path optimizer merges ironing lines with infill lines. Result of this the ironing skips some parts.
+                disable_path_optimisation = disable_path_optimisation || (mesh.settings.get<bool>("ironing_enabled") && !mesh.layers[gcode_layer.getLayerNr()].top_surface.areas.empty());
```

Comment:
This is not a good solution, since it disables path optimisation for the entire print if ironing is enabled in any mesh. This causes a lot of prints to become ugly, increase printing time by an order of magnitude in some cases, etc. That is actually also a problem with that Gyroid infill solution above.